### PR TITLE
Use ansible to deploy service catalog, ansible service broker, template service broker

### PR DESCRIPTION
**Pull request by maxamillion**
Signed-off-by: Adam Miller <admiller@redhat.com>

**What this PR does / why we need it**:
Enables OpenShift to re-use more deployment components from the official installer [openshift-ansible](https://github.com/openshift/openshift-ansible), specifically to deploy the service catalog, ansible service broker, template service broker, and examples.

### CODE REVIEW

File: pkg/openshift/certgen/templates/master/tmp/ansible/azure-local-master-inventory.yml [hunk 1]
```diff
@@ -64,6 +65,16 @@ localmaster:
     openshift_prometheus_image_prefix: "IMAGE_PREFIX/"
     openshift_deployment_type: 'ANSIBLE_DEPLOY_TYPE'
 
+    ansible_service_broker_install: True
+    openshift_service_catalog_image_prefix: "IMAGE_PREFIX/IMAGE_TYPE-"
+    openshift_service_catalog_image_version: 'vVERSION'
+    openshift_master_etcd_urls: ["https://HOSTNAME:2379"] # FIXME - not sure what this should be
```
Comment (jim-minter) [inline pkg/openshift/certgen/templates/master/tmp/ansible/azure-local-master-inventory.yml]: looks fine to me

File: pkg/openshift/certgen/templates/master/tmp/ansible/azure-ocp-deploy.yml [hunk 2]
```diff
@@ -21,6 +21,30 @@
       import_role:
         name: openshift_web_console
 
+    # FIXME - There's a race condition where the infra label isn't applied to
+    #         the infra node during cluster bring up in
+    #         openshift_bootstrap_autoapprover, this is problematic because the
+    #         openshift_prometheus role checks to ensure there is a node with
```
Comment (kargakis) [inline pkg/openshift/certgen/templates/master/tmp/ansible/azure-ocp-deploy.yml]: Is there an issue in openshift-ansible to track fixing this?

Comment (maxamillion) [reply pkg/openshift/certgen/templates/master/tmp/ansible/azure-ocp-deploy.yml]: Yes https://github.com/openshift/openshift-ansible/issues/8028

**Comment by jim-minter** [comment]
lgtm, @CecileRobertMichon please could you merge?

**Comment by jim-minter** [comment]
nice work @maxamillion, thanks!

**Comment by maxamillion** [comment]
Added examples, rebased and squashed

**Comment by jim-minter** [comment]
still lgtm, thanks

**Comment by maxamillion** [comment]
Apologies, had to fix a typo. I'm sure I just restarted the E2E clock.

**Comment by jim-minter** [comment]
@CecileRobertMichon please can you have another go at merging this?

**Comment by kargakis** [comment]
Seems like this PR missed checking in some generated code (diff can be found in https://github.com/Azure/acs-engine/pull/2746). @jim-minter @CecileRobertMichon shouldn't this be caught by testing?

**Pull request merged**